src/SignalR/clients/cpp/src/signalrclient/http_client.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <string>
+#include <functional>
+#include <map>
+#include <chrono>
+
+namespace signalr
+{
+    enum http_method
+    {
+        GET,
+        POST
+    };
+
+    class http_request
+    {
+    public:
+        http_method method;
+        std::map<std::string, std::string> headers;
+        std::string content;
+        std::chrono::seconds timeout;
+    };
+
+    class http_response
+    {
+    public:
+        int status_code = 0;
+        std::string content;
```

Comment:
Are we going to support SSE? If not, and I think we can probably do without it, this should be fine. If we do want to support it, we'll need this to be a stream of some kind.

Reply:
We just need to make sure that it can be added.

Reply:
We talked about using ostream for this and istream for the request content. The problem is that only offers blocking and polling APIs. The alternative is using [Asio's AsyncRead/WriteStream](http://think-async.com/Asio/asio-1.11.0/doc/asio/reference/AsyncWriteStream.html).

Since we're only aiming for source compatibility initially instead of some stable C ABI, we can probably just add an async stream field to http_request/response later and gracefully skip SSE if the http_client that doesn't use the field. We'd need a new send overload that calls the callback prior to buffering the entire response anyway.